src/modalities/nn/weight_init/low_level_weight_init_factory.py

```diff
@@ -0,0 +1,121 @@
+    ModulewiseNormalInitialization,
+    NamedParameterwiseNormalInitialization,
+    WeightInitializationIF,
+    WeightInitializerWrapper,
+)
+
+
+class WeightInitializerWrapperConfig(BaseModel):
+    weight_initializers: List[PydanticWeightInitializationIFType]
+
+
+class PlainWeightInitializationConfig(BaseModel):
+    mean: float
+    std: Annotated[float, Field(strict=True, ge=0.0)] | str  # can be float or "auto"
+    hidden_dim: Optional[int] = None
+
+    @model_validator(mode="after")
+    def check_std_and_hidden_dim(self):
+        if self.std == "auto" and self.hidden_dim is None:
+            raise ValueError("hidden_dim must be specified when std is 'auto'")
+        return self
+
+
+class NamedParameterwiseNormalInitializationConfig(BaseModel):
+    mean: float
+    std: Annotated[float, Field(strict=True, ge=0.0)] | str  # can be float or "auto"
+    parameter_name_suffixes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+
+
+class ScaledWeightInitializationConfig(BaseModel):
+    mean: float
+    plain_std: Annotated[float, Field(strict=True, ge=0.0)]
+    num_layers: Annotated[int, Field(strict=True, gt=0)]
+    parameter_name_suffixes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+
+
+class ScaledEmbedInitializationConfig(BaseModel):
+    mean: float
+    parameter_name_suffixes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+
+
+class LowLevelInitializationFactory:
+    @staticmethod
+    def get_weight_initializer_wrapper(weight_initializers: List[WeightInitializationIF]) -> WeightInitializationIF:
+        initializer_wrapper = WeightInitializerWrapper(weight_initializers)
+        return initializer_wrapper
+
+    @staticmethod
+    def get_plain_initialization(
+        mean: float, std: float | str, hidden_dim: Optional[int] = None
+    ) -> WeightInitializationIF:
+        """Initializes the weights of a model by sampling from a normal distribution.
+        NOTE: This class supports the initialization of nn.Linear and nn.Embedding layers.
+        For other layer types, the initialization must be subclassed and extended
+
+        Args:
+            mean (float): mean of the normal distribution
+            std (float): standard deviation of the normal distribution
+            hidden_dim (Optional[int], optional): hidden dimension of the attention layer. Defaults to None.
+        """
+
+        # auto: choose std automatically
+        if std == "auto":
+            if hidden_dim is None:
+                raise ValueError("ERROR! weight_init.std = auto not implemented")
+            # as per  https://arxiv.org/abs/2312.16903
+            std = math.sqrt(2 / (5 * hidden_dim))
+
+        initialization = ModulewiseNormalInitialization(mean=mean, std=std)
+        return initialization
+
+    @staticmethod
+    def get_scaled_initialization(
+        mean: float, plain_std: float, num_layers: int, parameter_name_suffixes: List[str]
```

Comment:
```suggestion
        mean: float, plain_std: float | str, num_layers: int, parameter_name_suffixes: List[str]
```
If `plain_std` is a str, line 97 will be ill-defined and cause an Error. That's why we need the check and the scaling (`std = math.sqrt(2 / (5 * hidden_dim))`) as for the `plain` initialization.

Reply:
resolved by https://github.com/Modalities/modalities/pull/168#discussion_r1663192270